pkg/provision/storage/shared.go

```diff
@@ -277,3 +277,32 @@ func checkPVCTerminating(name, namespace string, api sync.ClusterAPI) (bool, err
 	}
 	return pvc.DeletionTimestamp != nil, nil
 }
+
+func shouldPersistHomeDirectory(workspace *common.DevWorkspaceWithConfig) bool {
+	return workspace.Config.Workspace.PersistUserHome != nil && workspace.Config.Workspace.PersistUserHome.Enabled != nil && *workspace.Config.Workspace.PersistUserHome.Enabled
+}
```

Comment:
By using a default value, we can assume this will never be nil -- it's not possible to unset this value since it should get replaced by the default config.